packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php

```diff
@@ -0,0 +1,124 @@
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    getJson(route('admin.catalog.families.index'), [
+        'X-Requested-With' => 'XMLHttpRequest',
+    ])
+        ->assertOk()
+        ->assertJsonPath('records.0.id', 1)
+        ->assertJsonPath('records.0.code', 'default')
+        ->assertJsonPath('meta.total', 1);
+});
+
+it('should return attribute family create page', function () {
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    get(route('admin.catalog.families.create'))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.catalog.families.create.title'))
+        ->assertSeeText(trans('admin::app.catalog.families.create.back-btn'));
+});
+
+it('should store newly created attribute family', function () {
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    postJson(route('admin.catalog.families.store'), [
+        'code'             => $code = strtolower(fake()->words(1, true)),
+        'name'             => $name = fake()->name(),
+        'attribute_groups' => [
+            [
+                'code'   => $code,
+                'name'   => $name,
+                'column' => 1,
+            ],
+        ],
+    ])
+        ->assertRedirectToRoute('admin.catalog.families.index')
+        ->isRedirection();
+
+    $this->assertDatabaseHas('attribute_families', [
+        'code' => $code,
+        'name' => $name,
+    ]);
+});
+
+it('should return edit page of attribute families', function () {
+    // Arrange
+    $AttributeFamily = AttributeFamilyModel::factory()->create();
+
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    get(route('admin.catalog.families.edit', $AttributeFamily->id))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.catalog.families.edit.title'))
+        ->assertSeeText(trans('admin::app.catalog.families.edit.back-btn'));
+});
+
+it('should update the existing attribute families', function () {
+    // Arrange
+    $attributeFamily = AttributeFamilyModel::factory()->create();
+
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    putJson(route('admin.catalog.families.update', $attributeFamily->id), [
+        'code' => $updatedCode = strtolower(fake()->words(1, true)),
+        'name' => $attributeFamily->name,
+    ])
+        ->assertRedirectToRoute('admin.catalog.families.index')
+        ->isRedirection();
+
+    $this->assertDatabaseHas('attribute_families', [
+        'code' => $updatedCode,
+        'name' => $attributeFamily->name,
+    ]);
+});
+
+it('should update the existing attribute family', function () {
+    // Arrange
+    $attributeFamily = AttributeFamilyModel::factory()->create();
+
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    deleteJson(route('admin.catalog.families.delete', $attributeFamily->id))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.catalog.families.delete-success'));
+
+    $this->assertDatabaseMissing('attribute_families', [
+        'id' => $attributeFamily->id,
+    ]);
+});
```

Comment:
Add one more testcase here,

should not be able to delete the attribute family if the attribute family is the only one present

Reply:
done